ui/src/app/core/capabilities.service.ts

```diff
@@ -0,0 +1,25 @@
+import {Headers, Http, RequestOptions} from '@angular/http';
+import {Injectable} from '@angular/core';
+import 'rxjs/add/operator/toPromise';
+
+import {CapabilitiesResponse} from '../shared/model/CapabilitiesResponse';
+import {environment} from '../../environments/environment';
+
+/** Service wrapper for accessing the capabilities endpoint. */
+@Injectable()
+export class CapabilitiesService {
+
+  private capabilitiesResponse: Promise<CapabilitiesResponse>;
+
+  constructor(private http: Http) {}
+
+  getCapabilities(): Promise<CapabilitiesResponse> {
```

Comment:
Per offline discussion, I would also support a synchronous getter for this which can be used from components, to simplify the initialization life cycle (throw error if not populated).

Reply:
Done.